specs/publisher_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'rspec'
+require 'simplecov'
+SimpleCov.start
+require_relative '../lib/publisher'
+
+RSpec.describe Publisher do
+  context '#valid_queue_name?' do
+    let (:valid_queue_name) { "test_123" }
+    let (:invalid_queue_name1) { 'a'*81 }
+    let (:invalid_queue_name2) { 'testing/1@2' }
+    
+    it "should validate the queue name as per aws sqs format and return true for valid queue name" do
+      expect(Publisher.valid_queue_name?(valid_queue_name)).to be_truthy
+    end
+
+    it "should validate the queue name as per aws sqs format and raise error for invalid queue name" do
```

Comment:
Break this into 2 specs
```
[invalid_queue_name1, invalid_queue_name2].each do |invalid_queue_name|
  it "should validate the queue name as per aws sqs format and raise error for #{invalid_queue_name}" do
  ...
  ...
  end
end
```

Do a similar code for valid/invalid sleep period, queue urls